Add basic app drawer

bhavjitChauhan opened the pull request:
Added basic app drawer functionality. Applications displayed are only for demonstration purposes. The design is simple and not too complex as expanding the dock would have been.

bhavjitChauhan committed:
Add basic app drawer

+ Added App Drawer application
+ Added Dock Expand icon
+ Added Expand button to the taskbar
+ Fixed taskbar icons not aligned
+ Renamed `navigationHome` to `navigationHomeIcon`

athaun commented:
I think Michael will like this better anyways, looks great. although the background shifts up and left, did u do that on purpose? @bhavjitChauhan

bhavjitChauhan commented:
It's a side-effect of the blur filter. I'll try to use the translate function to fix it.

bhavjitChauhan committed:
Remove app drawer blur

Removed blur when app drawer is shown for the following reasons:
 - Caused inconsistent framerate drops when shown
 - Slightly translated the entire canvas into the left corner, translating within the app drawer will not solve the issue since the app drawer is drawn towards the end of the draw loop
 - Custom `blur` function in the Simple Elements library does not work well on the entire canvas

bhavjitChauhan committed:
Rework app drawer

+ Added morph animations from taskbar to app drawer
+ Added `setX` and `setY` methods to `Element` object
- Removed `appDrawer` app

bhavjitChauhan commented:
I've added the basic functionality for the app drawer again - this time with the morph animations. It doesn't look pretty so the graphics team will need to tweak it to make it look the part.

bhavjitChauhan committed:
Add documentation comments
